main.cpp

```diff
@@ -1,66 +1,76 @@
+* limitations under the License.
+*/
 #include "mbed.h"
 #include "mbed_error.h"
 #include "mbed_fault_handler.h"
 
-mbed_error_ctx error_ctx;
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+mbed_error_status_t err_status;
 mbed_fault_context_t fault_ctx;
-void generate_bus_fault_unaligned_access();
+static bool reboot_error_happened = false;
 
-static int reboot_error_happened = 0;
-static bool continuous_reboot_test = false;
+void generate_bus_fault_unaligned_access()
+{
+    SCB->CCR |= SCB_CCR_UNALIGN_TRP_Msk;
+    uint32_t inv_addr = 0xAAA3;
+    uint32_t val = *(uint32_t *)inv_addr;
+    printf("val= %X\n", (unsigned int)val);
+    return;
+}
 
-#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
-void mbed_error_reboot_callback(mbed_error_ctx *error_context) {
-    printf("\nmbed_error_reboot_callback: System rebooting, reboot error callback received");
-    reboot_error_happened = 1;
-    mbed_get_reboot_error_info(&error_ctx);
+//   Implementation for Mbed OS fatal error reboot callback
+void mbed_error_reboot_callback(mbed_error_ctx *error_context)
+{
+    reboot_error_happened = true;
+    err_status = error_context->error_status;
+    printf("\nError:\n");
+    printf("Status: 0x%lX\n", (uint32_t)error_context->error_status);
+    printf("Value: 0x%lX\n", (uint32_t)error_context->error_value);
+    printf("Address: 0x%lX\n", (uint32_t)error_context->error_address);
+    printf("Reboot count: 0x%lX\n", (uint32_t)error_context->error_reboot_count);
+    printf("CRC: 0x%lX\n", (uint32_t)error_context->crc_error_ctx);
     mbed_reset_reboot_error_info();
 }
 #endif
 
 // main() runs in its own thread in the OS
-int main() {
-#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED  
-    printf("\n\nMbed-OS crash reporting test: main()\n");
-    if((reboot_error_happened == 0) || continuous_reboot_test) {
-        printf("\nForcing exception\n");
+int main()
+{
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+    printf("This is the crash reporting Mbed OS example\n");
+    if (!reboot_error_happened) {
+        // Inject the fault exception
         generate_bus_fault_unaligned_access();
-        printf("\nForcing exception failed\n");
     } else {
-        if(error_ctx.error_status < 0) {
-            printf("\nSuccessfully read reboot info ==> \n");
-            printf("\n  error status: 0x%08lX", (uint32_t)error_ctx.error_status);
-            printf("\n  error value: 0x%08lX", (uint32_t)error_ctx.error_value);
-            printf("\n  error address: 0x%08lX", (uint32_t)error_ctx.error_address);
-            printf("\n  error reboot count: 0x%08lX", (uint32_t)error_ctx.error_reboot_count);
-            printf("\n  error crc: 0x%08lX\n", (uint32_t)error_ctx.crc_error_ctx);
-            
-            //Read fault context
-            if(error_ctx.error_status == MBED_ERROR_HARDFAULT_EXCEPTION) {
+        if (err_status < 0) {            
+            if (err_status == MBED_ERROR_HARDFAULT_EXCEPTION) {
```

Comment:
Why do we need to check for this error status?
If you check then print an error if the error status is not what you expect.